// <copyright file="PirateMessageTest.GetGameName.g.cs">Copyright ©  2011</copyright>
// <auto-generated>
// This file contains automatically generated unit tests.
// Do NOT modify this file manually.
// 
// When Pex is invoked again,
// it might remove or update any previously generated unit tests.
// 
// If the contents of this file becomes outdated, e.g. if it does not
// compile anymore, you may delete this file and invoke Pex again.
// </auto-generated>
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Microsoft.Pex.Framework.Generated;
using Microsoft.Pex.Engine.Exceptions;

namespace PirateSpades.Network
{
    public partial class PirateMessageTest {
[TestMethod]
[PexGeneratedBy(typeof(PirateMessageTest))]
[PexRaisedContractException(PexExceptionState.Expected)]
public void GetGameNameThrowsContractException629()
{
    try
    {
      PirateMessage pirateMessage;
      string s;
      pirateMessage =
        new PirateMessage(PirateMessageHead.Fail, "game_name: \0\0\0\0");
      s = this.GetGameName(pirateMessage);
      throw 
        new AssertFailedException("expected an exception of type ContractException");
    }
    catch(Exception ex)
    {
      if (!PexContract.IsContractException(ex))
        throw ex;
    }
}
[TestMethod]
[PexGeneratedBy(typeof(PirateMessageTest))]
[PexRaisedContractException(PexExceptionState.Expected)]
public void GetGameNameThrowsContractException331()
{
    try
    {
      PirateMessage pirateMessage;
      string s;
      pirateMessage = new PirateMessage(PirateMessageHead.Fail, "game_name: \0\0\0");
      s = this.GetGameName(pirateMessage);
      throw 
        new AssertFailedException("expected an exception of type ContractException");
    }
    catch(Exception ex)
    {
      if (!PexContract.IsContractException(ex))
        throw ex;
    }
}
[TestMethod]
[PexGeneratedBy(typeof(PirateMessageTest))]
public void GetGameName787()
{
    PirateMessage pirateMessage;
    string s;
    pirateMessage = new PirateMessage(PirateMessageHead.Fail, "game_name: A");
    s = this.GetGameName(pirateMessage);
    Assert.AreEqual<string>("A", s);
}
[TestMethod]
[PexGeneratedBy(typeof(PirateMessageTest))]
[PexRaisedContractException(PexExceptionState.Expected)]
public void GetGameNameThrowsContractException725()
{
    try
    {
      PirateMessage pirateMessage;
      string s;
      pirateMessage = new PirateMessage(PirateMessageHead.Fail, "game_name: \0\0");
      s = this.GetGameName(pirateMessage);
      throw 
        new AssertFailedException("expected an exception of type ContractException");
    }
    catch(Exception ex)
    {
      if (!PexContract.IsContractException(ex))
        throw ex;
    }
}
[TestMethod]
[PexGeneratedBy(typeof(PirateMessageTest))]
[PexRaisedContractException(PexExceptionState.Expected)]
public void GetGameNameThrowsContractException308()
{
    try
    {
      PirateMessage pirateMessage;
      string s;
      pirateMessage = new PirateMessage(PirateMessageHead.Fail, "game_name: \0");
      s = this.GetGameName(pirateMessage);
      throw 
        new AssertFailedException("expected an exception of type ContractException");
    }
    catch(Exception ex)
    {
      if (!PexContract.IsContractException(ex))
        throw ex;
    }
}
[TestMethod]
[PexGeneratedBy(typeof(PirateMessageTest))]
[PexRaisedContractException(PexExceptionState.Expected)]
public void GetGameNameThrowsContractException791()
{
    try
    {
      PirateMessage pirateMessage;
      string s;
      pirateMessage = new PirateMessage(PirateMessageHead.Fail, "");
      s = this.GetGameName(pirateMessage);
      throw 
        new AssertFailedException("expected an exception of type ContractException");
    }
    catch(Exception ex)
    {
      if (!PexContract.IsContractException(ex))
        throw ex;
    }
}
[TestMethod]
[PexGeneratedBy(typeof(PirateMessageTest))]
[PexRaisedContractException(PexExceptionState.Expected)]
public void GetGameNameThrowsContractException510()
{
    try
    {
      string s;
      s = this.GetGameName((PirateMessage)null);
      throw 
        new AssertFailedException("expected an exception of type ContractException");
    }
    catch(Exception ex)
    {
      if (!PexContract.IsContractException(ex))
        throw ex;
    }
}
    }
}
